src/test/java/de/tum/cit/aet/artemis/communication/notification/SingleUserNotificationServiceTest.java

```diff
@@ -503,6 +509,58 @@ void checkFirstNotification() {
         }
     }
 
+    // User VCS access token related (expiry warning and newly added token)
+
+    @Nested
+    class UserTokenExpiryNotification {
+
+        List<Notification> sentNotifications;
+
+        @AfterEach
+        void tearDown() throws Exception {
+            user.setVcsAccessTokenExpiryDate(null);
+            user.setVcsAccessToken(null);
+            userTestRepository.save(user);
+        }
+
+        @Test
+        void shouldNotifyUserAboutNewlyAddedVcsAccessToken() {
+            singleUserNotificationService.notifyUserAboutNewlyAddedVcsAccessToken(user);
+
+            sentNotifications = notificationRepository.findAll();
+            assertThat(sentNotifications.getFirst()).isInstanceOf(SingleUserNotification.class);
```

Comment:
This can cause flaky tests as this depends on an absolute state of the database, in particular for parallel execution. Could we maybe use the userId to fetch the notifications for that specific user?

Reply:
good idea, I added a testRepository for that purpose